### PR TITLE
Add Skill Form Component to Resume Builder

### LINKED ISSUE #9: Page for adding a new Skill
Add a page that allows users to add a new Skill which includes the following fields:

- name
- proficiency
- logo

Collaborate with the fellows contributing to the Python project to understand how logos will be managed. For now, assume this is a string (URL to the image). 

When this information is submitted, a POST request should be made to `/resume/skill` on the server in the following format:

```json
{
    "name": "JavaScript",
    "proficiency": "2-4 years",
    "logo": "example-logo.png"
}
```

The server will return a JSON with the position in the list in the following format:
```json
{
"id": 1
}
```

### PR DESCRIPTION
**Pull request by esohel30**
close #9

This PR introduces a new feature to the Resume Builder application by adding a form for users to input and submit their skills. The following changes have been made:

**AddSkillForm Component:**
1. Created AddSkillForm.js which includes a form for adding a new skill with fields for skill name, proficiency, and logo URL.
2. Implemented async/await for handling the POST request to the server.
3. Added error handling for network requests.
4. Included onSubmit prop to handle actions after form submission.
5. Form fields are marked as required to ensure data completeness.

**App Component:**
1. Updated App.js to include and toggle the AddSkillForm component.
2. Added logic to display the AddSkillForm when the "Add Skill" button is clicked and hide it after submission.

Styling:
1. Created AddSkillForm.css to style the AddSkillForm component with basic form styling including input fields and submit button.

Screenshots

Screen Shot 2024-05-27 at 9 31 28 PM

### CODE REVIEW

**Comment by esohel30** [comment]
@Pradyuman7 tagging you here as I cannot assign you as a reviewer.

File: src/AddSkillForm.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(skillData),
+      });
+
+      const result = await response.json();
+
+      if (response.ok) {
+        console.log(`Skill added with ID: ${result.id}`);
```
Comment (Pradyuman7) [inline src/AddSkillForm.js]: nit: we don't need to show the id to the user here

Comment (esohel30) [reply src/AddSkillForm.js]: @Pradyuman7 Fixed this problem.

**Review by Pradyuman7 — COMMENTED** [review]
LGTM 👍 

Can you resolve the conflicts, I'll merge then

**Comment by esohel30** [comment]
@Pradyuman7 LGTM 🚀 conflicts resolved.

**Review by Pradyuman7 — APPROVED** [review]
LGTM 👍 

one final conflict, can you please resolve it. I'll merge the PR then

**Comment by esohel30** [comment]
@Pradyuman7 fixed and ready to merge boys💯💯💯💯

**Pull request merged**